### PR TITLE
simplify UI and exposed Data model

### PR DESCRIPTION
**Pull request by ndeloof**
as of SECURITY-440 only directly entered private key is supported, others legacy flavours are automatically migrated. This PR do simplify the UI to only expose this directly entered key as a single textarea field.
It also make exposed data model cleaner, typically configuration-as-code would avoid a deep nested model to define private key

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java [hunk 1]
```diff
@@ -100,11 +101,22 @@
...
      * @param scope            the credentials scope
      * @param username         the username.
-     * @param privateKeySource the private key.
+     * @param privateKey       the private key.
      * @param passphrase       the password.
      * @param description      the description.
      */
     @DataBoundConstructor
+    public BasicSSHUserPrivateKey(CredentialsScope scope, String id, String username, String privateKey,
+                                  String passphrase,
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java]: agree, didn't want to make more changes that strictly required, but I'm fine to get one step further

File: src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java [hunk 2]
```diff
@@ -26,6 +26,7 @@
 import com.cloudbees.jenkins.plugins.sshcredentials.SSHUserPrivateKey;
 import com.cloudbees.plugins.credentials.CredentialsProvider;
 import com.cloudbees.plugins.credentials.CredentialsScope;
+import com.google.common.base.Optional;
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java]: my bad

File: src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java [hunk 3]
```diff
@@ -100,11 +101,22 @@
...
+                                  String passphrase,
+                                  String description) {
+        super(scope, id, username, description);
+        privateKey = StringUtils.trimToNull(privateKey);
+        this.privateKeySource = privateKey == null ? null : new DirectEntryPrivateKeySource(privateKey);
+        this.passphrase = fixEmpty(passphrase == null ? null : Secret.fromString(passphrase));
+    }
+
+
+    @Deprecated
```
[the remark this answers is not in the document]
Comment (ndeloof) [reply src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java]: ack

File: src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndeloof) [reply src/main/java/com/cloudbees/jenkins/plugins/sshcredentials/impl/BasicSSHUserPrivateKey.java]: why use a DataBoundSetter ? Does it make any sense to create a SSHUserPrivateKey without a private key ?

**Comment by jglick** [comment]
BTW merge with `master` to make CI builds work.

**Comment by casz** [comment]
@jvz has this been replaced by #39 and #40 ?

**Comment by jvz** [comment]
Thanks, you're right.

**Pull request closed without merging**